loader/loader.go

```diff
@@ -139,7 +141,8 @@ func (l *loader) GetComponentFromSnapshot(c client.Client, ctx context.Context,
 
 		return component, nil
 	} else {
-		return nil, fmt.Errorf("%s label missing from the Snapshot", gitops.SnapshotComponentLabel)
+		groupResource := schema.GroupResource{Group: "", Resource: ""}
+		return nil, errors.NewNotFound(groupResource, fmt.Sprintf("%s label missing from the Snapshot", gitops.SnapshotComponentLabel))
```

Comment:
I'm kinda worried of this sending the wrong message about a missing Component that's referenced by the Snapshot, instead of the Snapshot's label that's actually missing.

Reply:
I've modified the HandleLoaderError() to also log the `err`, and it will clarify whether the Component is not found or the component label is not found. My cluster died or else I would've pasted the latest logs for both scenarios. Will attach them here in 30 mins.

Reply:
Here's what the log looks like when the Component is missing from the cluster:
```
{"level":"info","ts":"2024-03-12T12:43:26Z","logger":"controllers.snapshot","caller":"helpers/errorhandlers.go:70","msg":"Could not get Component or 'appstudio.openshift.io/component' label from Snapshot. Component or 'appstudio.openshift.io/component' label may have been removed. Declining to proceed with reconciliation due to the error: Component.appstudio.redhat.com \"component-sample\" not found","snapshot":{"name":"simple-nodejs-app-2zfkc","namespace":"default"},"application":"default/application-sample"}
```

Here's what the log looks like when the component label is missing from the Snapshot:
```
{"level":"info","ts":"2024-03-12T12:43:56Z","logger":"controllers.snapshot","caller":"helpers/errorhandlers.go:70","msg":"Could not get Component or 'appstudio.openshift.io/component' label from Snapshot. Component or 'appstudio.openshift.io/component' label may have been removed. Declining to proceed with reconciliation due to the error: \"Label 'appstudio.openshift.io/component'\" not found","snapshot":{"name":"simple-nodejs-app-djvpp","namespace":"default"},"application":"default/application-sample"}
```